packages/mui-joy/src/CircularProgress/CircularProgressCssVars.ts

```diff
@@ -0,0 +1,18 @@
+import { GenerateCssVarsType } from '../utils/GenerateCssVarsType';
+
+const circularProgressCssVars = {
+  thickness: '--CircularProgress-thickness',
+  trackThickness: '--CircularProgress-trackThickness',
+  progressThickness: '--CircularProgress-progressThickness',
+  trackColor: '--CircularProgress-trackColor',
+  progressColor: '--CircularProgress-progressColor',
+  percent: '--CircularProgress-percent',
+  linecap: '--CircularProgress-linecap',
+  size: '--CircularProgress-size',
+  margin: '--CircularProgress-margin',
+  circulation: '--CircularProgress-circulation',
+} as const;
```

Comment:
Suggestion: we can have a utility function similar to https://vanilla-extract.style/documentation/global-api/create-global-theme-contract/ that will generate the variables object with values equal to the css variable string at any level of nesting instead of manually writing the whole string.